include/seqan3/range/views/kmer_hash.hpp

```diff
@@ -294,7 +324,7 @@ class kmer_hash_view<urng_t>::shape_iterator
     {
         assert(std::ranges::size(shape_) > 0);
 
-        if (shape_.size() <= std::ranges::distance(text_left, text_right))
+        if (shape_.size() <= std::ranges::distance(text_left, text_right) + 1)
```

Comment:
Why this change?

Reply:
Basically:

```
        // shape size = 3
        // Text:     1 2 3 4 5 6 7 8 9
        // it_start: ^
        // it_end:       ^
        // distance(it_start, it_end) = 2
```

But I'm still investigating other stuff.